### PR TITLE
feat(core): add parseMinMergeScore helper

### PR DESCRIPTION
**Pull request by joshuaswarren**
Part of #2330

## Change
parseMinMergeScore. Finite number in [0,1]. Outside range throws.

## Test
packages/remnic-core/src/dedup/merge-min-score.test.ts

## Summary by CodeRabbit

* **New Features**
  * Added validation for merge-on-write minimum scores.
  * Accepts finite numeric values from 0 through 1, including boundary values.

* **Bug Fixes**
  * Invalid, non-numeric, out-of-range, or non-finite score values now produce clear validation errors.

* **Release**
  * Included a patch release update for the core package.

### CODE REVIEW

File: packages/remnic-core/src/dedup/merge-min-score.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Parse merge-on-write min score (issue #2330 leftover).
+ *
+ * Finite number in [0, 1] is returned. Outside range or non-finite throws.
+ */
+export function parseMinMergeScore(value: unknown): number {
+  if (typeof value !== "number" || !Number.isFinite(value) || value < 0 || value > 1) {
+    throw new RangeError(
+      `merge minScore must be a finite number in [0, 1]; got ${JSON.stringify(value)}`,
+    );
```
[the remark this answers is not in the document]
Comment (joshuaswarren) [reply packages/remnic-core/src/dedup/merge-min-score.ts:19]: Fixed in c77e6e10a: replaced `JSON.stringify` with a non-throwing formatter that falls back to `String(value)` on bigint/cyclic inputs. Added regression tests for both cases (7/7 pass).
